2018/gabriel_python3/src/day11/11.2.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+import sys
+
+def get_hundredth_digit(number):
+    try:
+        return int(str(number)[-3])
+    except IndexError:
+        return 0
+
+def power_cell(serial, x, y):
+    rack_id = x + 10
+    power_level = rack_id * y
+    power = power_level + serial
+    hundred = get_hundredth_digit(power * rack_id)
+    return hundred - 5
+
+
+def area_power(grid, x, y, size):
+    # TODO: replace with incremental sum
+    power = 0
+    for i in range(size):
+        for j in range(size):
+            power += grid[x + i][y + j]
+    return power
+
+
+def puzzle(serial_number):
+    grid = [0] * 300
+    for i in range(300):
+        grid[i] = [0] * 300
+    
+    for x in range(300):
+        for y in range(300):
+            grid[x][y] = power_cell(serial_number, x + 1, y + 1)
+
+    max_power = 1 - sys.maxsize
+    max_coordinates = (None, None)
+    max_size = None
+    for x in range(0, 300):
+        for y in range(0, 300):
+            for size in range(1, 25):  # score decreases after 25
```

Comment:
That may be unfair to compare energy consumption with another solutions if you truncate range of size

Reply:
I don't think it will matter so much, but feel free to change it back to 300